backend/src/test/controller/EnumController.test.ts

```diff
@@ -1,21 +1,25 @@
 import connection from '../repository/BaseRepositoryTest';
 import { Express } from 'express';
 import request from 'supertest';
-import { registerTestApplication } from './BaseController';
+import { connectRedisClient, registerTestApplication } from './BaseController';
 import { DiscountType } from '../../main/data/DiscountType';
 import { PromotionType } from '../../main/data/PromotionType';
 import { CuisineType } from '../../main/data/CuisineType';
+import { RedisClient } from 'redis-mock';
 
 describe('Unit tests for PromotionController', function () {
   let app: Express;
+  let redisClient: RedisClient;
 
   beforeAll(async () => {
     await connection.create();
-    app = registerTestApplication();
+    redisClient = await connectRedisClient();
+    app = await registerTestApplication(redisClient);
   });
 
   afterAll(async () => {
     await connection.close();
+    redisClient.quit();
   });
 
```

Comment:
Yeah now the only con I think with this is that we need to use `redis` even for controller tests that don't involve redis. What would happen if we made `redisClient` an optional parameter to `registerTestApplication` and left the class the way it was before?